test/export/StructureDefinitionExporter.test.ts

```diff
@@ -609,6 +609,120 @@ describe('StructureDefinitionExporter', () => {
+    const valuesetExpression = sd.elements.find(
+      e => e.id === 'Observation.modifierExtension:valueset-expression'
+    );
+
+    expect(extension.slicing).toBeDefined();
+    expect(extension.slicing.discriminator.length).toBe(1);
+    expect(extension.slicing.discriminator[0]).toEqual({ type: 'value', path: 'url' });
+    expect(valuesetExpression).toBeDefined();
+    expect(valuesetExpression.type[0]).toEqual({
+      code: 'Extension',
+      profile: ['http://hl7.org/fhir/StructureDefinition/valueset-expression']
+    });
+  });
+
+  it('should apply a ContainsRule of an undefined extension on an extension element', () => {
+    const profile = new Profile('Foo');
+    profile.parent = 'Observation';
+
+    const rule = new ContainsRule('extension');
+    rule.items = ['foo'];
+    profile.rules.push(rule);
+
+    exporter.exportStructDef(profile);
+    const sd = exporter.structDefs[0];
+
+    const extension = sd.elements.find(e => e.id === 'Observation.extension');
+    const foo = sd.elements.find(e => e.id === 'Observation.extension:foo');
+
+    expect(extension.slicing).toBeDefined();
+    expect(extension.slicing.discriminator.length).toBe(1);
+    expect(extension.slicing.discriminator[0]).toEqual({ type: 'value', path: 'url' });
+    expect(foo).toBeDefined();
+    expect(foo.type[0]).toEqual({ code: 'Extension' });
+  });
+
+  it('should apply a ContainsRule of an aliased extension on an extension element', () => {
+    const profile = new Profile('Foo');
+    const extBar = new Extension('Bar');
+    const extBaz = new Extension('Baz');
+    extBaz.id = 'BazId';
+    profile.parent = 'Observation';
+
+    doc.aliases.set('barAlias', 'Bar');
+    doc.aliases.set('bazAlias', 'BazId');
+    doc.extensions.set('Bar', extBar);
+    doc.extensions.set('Baz', extBaz);
+
+    const ruleBar = new ContainsRule('extension');
+    ruleBar.items = ['barAlias'];
+    profile.rules.push(ruleBar);
+    const ruleBaz = new ContainsRule('extension');
+    ruleBaz.items = ['bazAlias'];
+    profile.rules.push(ruleBaz);
+
+    exporter.exportStructDef(profile);
+    const sd = exporter.structDefs[2];
+
+    const extension = sd.elements.find(e => e.id === 'Observation.extension');
+    const bar = sd.elements.find(e => e.id === 'Observation.extension:barAlias');
+    const baz = sd.elements.find(e => e.id === 'Observation.extension:bazAlias');
+
+    expect(extension.slicing).toBeDefined();
+    expect(extension.slicing.discriminator.length).toBe(1);
+    expect(extension.slicing.discriminator[0]).toEqual({ type: 'value', path: 'url' });
+    expect(bar).toBeDefined();
+    expect(bar.type[0]).toEqual({
+      code: 'Extension',
+      profile: ['http://example.com/StructureDefinition/Bar']
+    });
+    expect(baz).toBeDefined();
+    expect(baz.type[0]).toEqual({
+      code: 'Extension',
+      profile: ['http://example.com/StructureDefinition/BazId']
+    });
+  });
```

Comment:
Can you also add a test where the alias points to a URL for an existing FHIR extension?  That's the way I would expect alias to usually be used in this context.

Reply:
Ah good point! I will add that.